Workflows - Move variables and params info to their own top-level page (#25416)
DataDog/documentation: content/en/service_management/workflows/variables.md

To reference the input parameter in a step, use the syntax `{{ Trigger.<parameter name>}}`. For example, to reference an input parameter named `user`, use `{{Trigger.user}}`.

The **Input Parameters** section displays the names of all existing input parameters together with a counter. Hover over a counter to see which steps are using the parameter.

{{< img src="service_management/workflows/input-parameter3.png" alt="Hover over a counter to see which steps are using the parameter." style="width:60%;">}}

You can add an implicit input parameter (a parameter that doesn't already exist in the workflow) by typing it into a workflow step using the `{{ Trigger.<parameter name> }}` syntax. The next time you save the workflow, a dialog appears allowing you to convert the parameter to an explicit parameter. For more information on triggering workflows, see [Trigger a workflow][5].

If you're looking for an existing input parameter, start typing `{{ Trigger.` to see if it appears as a suggestion. Alternatively, consult the [Context Variables](#context-variables) tab for a list of available parameters.

## Output parameters

> You can pass the result of a workflow using output parameters.

Reviewer: ```suggestion
Output parameters allow you to access the result of a workflow. This is useful when you want to pass the result of a workflow to another workflow or an App Builder app.
```

To add an output parameter:
1. Click on the workflow canvas.
1. Click the **+** icon next to **Output Parameters**.
1. Add a parameter name, value, and data type for the parameter.
1. Optionally, add a default value for the parameter. If you add a default value, the parameter is optional at runtime.

The **Output Parameters** section displays the names of all existing output parameters together with a counter.

> 

Reviewer: ```suggestion
For information on using output parameters with workflows, see [Trigger a workflow from a workflow][7].
```
Need to add a small section to this page for that. I can follow up with a commit

For an example of how to use output parameters to pass information between Workflows and App Builder, see [return workflow results to an app][6].

## Custom variables

> To set a custom mutable workflow variable, use the [Set variable][1] action. You can use this action to declare, update, and access variables throughout your workflow, which allows you to perform more complex workflow operations. For example:

Reviewer: ```suggestion
To set a mutable workflow variable, use the [Set variable][1] action. You can use this action to declare, update, and access custom variables throughout your workflow, which allows you to perform more complex workflow operations. For example:
```

- _Handling API pagination_: API requests sometimes require you to keep track of a page token or offset.
- _Handling lists_: You can use a variable to initialize an array and perform actions like map and reduce.
- _Iteration_: Variables allow you to manipulate and store data inside a [for loop][2]. You can then use that data in the rest of the workflow.

> ### Set a variable

Reviewer: ```suggestion
### Set a custom variable
```

To avoid a type error resulting from an undefined variable, assign a custom variable before you use it in a loop. In the example below, the custom variable `evenList` is set to an empty array before it is used in the loop.

{{< img src="service_management/workflows/loop.png" alt="This workflow sets a variable before it is used in a loop." style="width:100%;" >}}

## Further reading

{{< partial name="whats-next/whats-next.html" >}}

<br>Do you have questions or feedback? Join the **#workflows** channel on the [Datadog Community Slack][3].

[1]: https://app.datadoghq.com/workflow/action-catalog#/com.datadoghq.core/com.datadoghq.core.setVariable
[2]: /service_management/workflows/actions/flow_control#for-loop
[3]: https://datadoghq.slack.com/
[4]: https://handlebarsjs.com/guide/expressions.html#expressions
[5]: /service_management/workflows/trigger

> [6]: /service_management/app_builder/queries/#return-workflow-results-to-an-app

Reviewer: ```suggestion
[6]: /service_management/app_builder/queries/#return-workflow-results-to-an-app
[7]: /service_management/workflows/trigger/#trigger-a-workflow-from-a-workflow
```
